src/handlers/canonicalFieldsHandlers.ts

```diff
@@ -1,13 +1,37 @@
+import { getLogger } from '../logger';
+import { db } from '../database';
+import { isCanonicalFieldArray } from '../types';
+import { ValidationError } from '../errors';
 import type {
   Request,
   Response,
 } from 'express';
 
-function getCanonicalFields(req: Request, res: Response): void {
-  res.status(200)
-    .contentType('application/json')
-    .send('{ "hello": "world"}');
-}
+const logger = getLogger(__filename);
+
+const getCanonicalFields = (req: Request, res: Response): void => {
+  db.sql('canonicalFields.fetchAll')
+    .then((canonicalFields) => {
+      logger.debug(canonicalFields, 'canonicalFields.fetchAll');
+      const { rows } = canonicalFields;
+      if (isCanonicalFieldArray(rows)) {
+        res.status(200)
+          .contentType('application/json')
+          .send(rows);
+      } else {
+        throw new ValidationError(
+          'The database responded with an unexpected format.',
+          isCanonicalFieldArray.errors,
+        );
+      }
+    })
+    .catch((error: unknown) => {
```

Comment:
I wonder what the default behavior of express is here and if we can define this behavior once in some other, higher-level code that we hook into express.

Reply:
I was curious about this question!

So it looks like there is default middleware that will handle and process errors: https://expressjs.com/en/guide/error-handling.html

Given that I think I should ditch this try / catch.

We might consider writing our own middleware some day, but at the very least I think it's clear that it's better to let errors bubble up to the middleware.

Reply:
Oops, I'm sort of wrong -- so that is all true for errors thrown, but failed promises are a different situation (promises that fail do so outside of the stack and need to be handled directly).

This means that we DO need a `.catch` but what might be better is to pass the error to `next` so the error is given to that earlier mentioned middleware.

I'm gonna switch over to the `pass error to next` approach

Reply:
Final update... sorry for the noise.

It turns out the express middleware, unfortunately, does not send the error that was thrown.  Given that I think maybe the current approach is "right" for this PR, but I'm going to make an issue to think more deeply about how we want our error handling to work.

Reply:
marking this as closed now that we have https://github.com/PhilanthropyDataCommons/service/issues/61

Reply:
Nice, yes. I think the `json` response even on 500 is nice and it doesn't look like that would happen with the default handler anyway. I also think logging the error is nice too. So yes, #61 is the perfect place to pursue something like this!